core/src/impl/Kokkos_Memory_Fence.hpp

```diff
@@ -55,7 +55,7 @@ void memory_fence() {
   __threadfence();
 #elif defined(KOKKOS_ENABLE_ROCM_ATOMICS)
   amp_barrier(CLK_LOCAL_MEM_FENCE | CLK_GLOBAL_MEM_FENCE);
-#elif defined(KOKKOS_ENABLE_HIP_ATOMICS)
+#elif defined(__HIP_DEVICE_COMPILE__)
```

Comment:
Hmm... this is at least inconsistent with the other definitions here. Does this make a difference?

Reply:
This one is a bit strange, 

`__HIP_DEVICE_COMPILE__ := __CUDA_ARCH__`

on the HIP/NVCC path -- more generally, it's the PP-define we use for "this is the device path".

I'm honestly not sure why these other ones are using the `KOKKOS_ENABLE_ROCM_ATOMICS` et al. to determine whether we have `__threadfence()`.

The reason this specifically had to change is that I am defining `KOKKOS_ENABLE_HIP_ATOMICS` on both host and device passes such that the function definitions exists on both passed (and therefore the AST matches), see the change to core/src/Kokkos_Atomic.hpp

Reply:
This one is a bit strange,

__HIP_DEVICE_COMPILE__ := __CUDA_ARCH__

on the HIP/NVCC path -- more generally, it's the PP-define we use for "this is the device path".

I'm honestly not sure why these other ones are using the KOKKOS_ENABLE_ROCM_ATOMICS et al. to determine whether we have __threadfence().

The reason this specifically had to change is that I am defining KOKKOS_ENABLE_HIP_ATOMICS on both host and device passes such that the function definitions exists on both passed (and therefore the AST matches), see the change to core/src/Kokkos_Atomic.hpp

Reply:
Yes, that's fine.